server/app/controllers/products.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-underscore-dangle */
+const ErrorResponse = require('../utils/errorResponse');
+const asyncHandler = require('../middleware/async');
+
+const List = require('../models/List');
+const Product = require('../models/Product');
+
+/**
+ * @api {post} /api/v1/lists/:listId/products
+ * @apiName createProduct
+ * @apiGroup products
+ * @apiPermission protected
+ *
+ * @apiDescription Create a product entry
+ */
+const createProduct = asyncHandler(async (req, res, next) => {
+  const { listId } = req.params;
+  const { payload } = req.body;
```

Comment:
Creating a new product requires two arguments: link and the listId. The `payload` should be `link`.